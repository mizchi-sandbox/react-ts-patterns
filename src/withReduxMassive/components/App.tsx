import * as React from "react"
import { Provider } from "react-redux"
import createStore from "../store/createStore"
import Greeting from "./atoms/Greeting"
import Counter from "./organisms/counter"

export default () => (
  <Provider store={createStore()}>
    <Counter />
  </Provider>
)
