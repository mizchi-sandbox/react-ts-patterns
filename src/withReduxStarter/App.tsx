import * as React from "react"
import { connect } from "react-redux"
import { increment, State } from "./store"

// connect
const mapStateToProps = (state: State) => {
  return {
    value: state.value
  }
}

type Props = {
  value: number
  increment(): void
}

export default connect(mapStateToProps, { increment })((props: Props) => {
  return (
    <div>
      <h1>Counter: Simple</h1>
      <div>
        value: {props.value}
        <button onClick={_ => props.increment()}>inc</button>
      </div>
    </div>
  )
})
