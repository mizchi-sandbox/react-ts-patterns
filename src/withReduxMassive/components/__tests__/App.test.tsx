import * as React from 'react'
import * as renderer from 'react-test-renderer'
import App from '../App'

test('render <App/>', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
