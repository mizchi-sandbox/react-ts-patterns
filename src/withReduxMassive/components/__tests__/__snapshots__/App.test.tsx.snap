// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <App/> 1`] = `
<div
  className="sc-bwzfXH bRhOHE"
>
  <div
    className="sc-htpNat hFgGyf"
  >
    <h1
      className="sc-bxivhb bOYIrQ"
    >
      Counter
    </h1>
    <span
      className="sc-ifAKCX cicvxl"
    >
      value: 
      0
    </span>
    <div
      className="sc-EHOje iafsfZ"
    >
      <button
        className="sc-bdVaJa ghfaWw"
        onClick={[Function]}
      >
        add:3
      </button>
      Â 
      <button
        className="sc-bdVaJa ghfaWw"
        onClick={[Function]}
      >
        increment
      </button>
    </div>
  </div>
</div>
`;
